src/core/Dropdown/index.tsx

```diff
@@ -114,7 +124,11 @@ const Dropdown = <Multiple extends boolean | undefined = false>({
   }, [value]);
 
   useEffect(() => {
-    if (isControlled) {
+    if (
+      isControlled &&
+      ((!multiple && propValue !== value) ||
+        (multiple && !compareOptionArrays(propValue, value)))
+    ) {
       setValue(propValue);
     }
   }, [propValue]);
```

Comment:
I'm not entirely sure whether this equality check should go here or above in the `value` useEffect.  That one may make more sense...